docs/reference/builder.md

```diff
@@ -10,48 +10,42 @@ parent = "mn_reference"
+The [`docker build`](/reference/commandline/build) command builds an image from a `Dockerfile` and a *context*. The build's context is the files at a specified location `PATH` or `URL`.  The `PATH` is a directory on your local filesystem.  The `URL` is a the location of a Git repository.
 
-Then call `docker build` with the path of your source repository as the argument
-(for example, `.`):
+A context is processed recursively. So, a `PATH` includes any subdirectories and the `URL` includes the repository and its submodules.  A simple build command that uses the current directory as context:
 
     $ docker build .
+    Sending build context to Docker daemon  6.51 MB
+    Sending build context to Docker daemon 
+    ...
+
+The build is run by the Docker daemon, not by the CLI. The first thing a build process does is send the entire context (recursively) to the daemon.  In most cases, it's best to start with empty directory as context and keep
+your Dockerfile in that directory. Add only the files needed for
+building the Dockerfile.
+
+>**Warning**: Do not use your root directory, `/`, as the `PATH` as it causes
+>the build to transfer the entire contents of your hard drive to the Docker
+>daemon. 
+
+To use a file in the build context, the `Dockerfile` can refer to it with
+command, for example,  a `COPY` instruction. To increase the build's
+performance, exclude files and directories by adding a `.dockerignore` file to
+the context directory.  For information about how to [create a `.dockerignore`
+file](#dockerignore-file) on this page.
 
-The path to the source repository defines where to find the *context* of
-the build. The build is run by the Docker daemon, not by the CLI, so the
-whole context must be transferred to the daemon. The Docker CLI reports
-"Sending build context to Docker daemon" when the context is sent to the daemon.
-
-> **Warning**
-> Avoid using your root directory, `/`, as the root of the source repository. The 
-> `docker build` command will use whatever directory contains the Dockerfile as the build
-> context (including all of its subdirectories). The build context will be sent to the
-> Docker daemon before building the image, which means if you use `/` as the source
-> repository, the entire contents of your hard drive will get sent to the daemon (and
-> thus to the machine running the daemon). You probably don't want that.
-
-In most cases, it's best to put each Dockerfile in an empty directory. Then,
-only add the files needed for building the Dockerfile to the directory. To
-increase the build's performance, you can exclude files and directories by
-adding a `.dockerignore` file to the directory.  For information about how to
-[create a `.dockerignore` file](#dockerignore-file) on this page.
+Traditionally, the `Dockerfile` is called `Dockerfile` and located in the root of the context. This isn't a requirement though.  You use the `-f` flag with `docker build` to point to a file anywhere in your file sytem.
```

Comment:
True, but the Dockerfile must be within the build-context, e.g.;

```
tree ../
../
├── Dockerfile
└── sub
    └── helloworld
```

From within the "sub" directory;

```Bash
$ docker build -t foo -f ../Dockerfile .
unable to prepare context: The Dockerfile (/home/example/Dockerfile) must be within the build context (.)
```

But this works;
```Bash
$ docker build -t foo -f ../Dockerfile ../
Sending build context to Docker daemon 3.584 kB
Step 0 : FROM ubuntu:latest
 ---> 6d4946999d4f
Step 1 : COPY helloworld /
helloworld: no such file or directory
```

(apart from the "helloworld" file not being found here)